library/net_sockets.c

```diff
@@ -270,13 +270,18 @@ static int net_would_block( const mbedtls_net_context *ctx )
  */
 static int net_would_block( const mbedtls_net_context *ctx )
 {
+    int err = errno;
+    
     /*
      * Never return 'WOULD BLOCK' on a non-blocking socket
      */
     if( ( fcntl( ctx->fd, F_GETFL ) & O_NONBLOCK ) != O_NONBLOCK )
+    {
+        errno = err;
         return( 0 );
+    }
 
-    switch( errno )
+    switch( errno = err )
```

Comment:
Its nothing urgent, basically the same issue that you are trying to fix in this function. At places, such as in lines 479-483, it is assumed that the call to `net_would_block()` does not modify the `errno`, but clearly there is no guarantee about this.